framework/src/meshgenerators/PatchMeshGenerator.C

```diff
@@ -0,0 +1,381 @@
+        node_positions.push_back({0.5 * (node_positions[i](0) + node_positions[(i + 1) % 4](0)),
+                                  0.5 * (node_positions[i](1) + node_positions[(i + 1) % 4](1)),
+                                  0.5 * (node_positions[i](2) + node_positions[(i + 1) % 4](2))});
+
+      // Exterior to interior midside node positions
+      for (unsigned i = 0; i < 4; ++i)
+        node_positions.push_back({0.5 * (node_positions[i](0) + node_positions[i + 4](0)),
+                                  0.5 * (node_positions[i](1) + node_positions[i + 4](1)),
+                                  0.5 * (node_positions[i](2) + node_positions[i + 4](2))});
+
+      // Interior midside node positions
+      for (unsigned i = 4; i < 8; ++i)
+        node_positions.push_back(
+            {0.5 * (node_positions[i](0) + node_positions[(i + 1) % 4 + 4](0)),
+             0.5 * (node_positions[i](1) + node_positions[(i + 1) % 4 + 4](1)),
+             0.5 * (node_positions[i](2) + node_positions[(i + 1) % 4 + 4](2))});
+    }
+
+    for (unsigned node_id = 0; node_id < num_nodes; ++node_id)
+      nodes[node_id] = mesh->add_point(node_positions[node_id], node_id);
+
+    if (_elem_type == "QUAD4")
+      makeQuad4Elems(*mesh, nodes);
+    else
+      makeQuad8Elems(*mesh, nodes);
+
+    boundary_info.sideset_name(1) = "bottom";
+    boundary_info.sideset_name(2) = "right";
+    boundary_info.sideset_name(3) = "top";
+    boundary_info.sideset_name(4) = "left";
+
+    boundary_info.nodeset_name(100) = "bottom_left";
+    boundary_info.nodeset_name(101) = "bottom_right";
+    boundary_info.nodeset_name(102) = "top_right";
+    boundary_info.nodeset_name(103) = "top_left";
+  }
+  else
+  {
+    const std::vector<Real> x_interior_fractions{
+        0.249, 0.826, 0.850, 0.273, 0.320, 0.677, 0.788, 0.165};
+    const std::vector<Real> y_interior_fractions{
+        0.342, 0.288, 0.649, 0.750, 0.186, 0.305, 0.693, 0.745};
+    const std::vector<Real> z_interior_fractions{
+        0.192, 0.288, 0.263, 0.230, 0.643, 0.683, 0.644, 0.702};
+
+    // Exterior node positions
+    std::vector<Point> node_positions{{_xoffset, _yoffset, _zoffset},
+                                      {xmax, _yoffset, _zoffset},
+                                      {xmax, ymax, _zoffset},
+                                      {_xoffset, ymax, _zoffset},
+                                      {_xoffset, _yoffset, zmax},
+                                      {xmax, _yoffset, zmax},
+                                      {xmax, ymax, zmax},
+                                      {_xoffset, ymax, zmax}};
+
+    // Interior node positions
+    for (unsigned i = 0; i < x_interior_fractions.size(); ++i)
+      node_positions.push_back({_xoffset + x_interior_fractions[i] * _xlength,
+                                _yoffset + y_interior_fractions[i] * _ylength,
+                                _zoffset + z_interior_fractions[i] * _zlength});
```

Comment:
Even for this kind of arithmetic you can make this a one liner if you put your lengths into a RealVectorValue and then just do the Vector operators on your Points. That's clean up a lot of this code.